# loading vector its more 5 gb in ram

In [3]:
from gensim.models.wrappers import FastText
from gensim.models import KeyedVectors

filename = 'D:/dataset/Embedding/Fastext/wiki-news-300d-1M.vec'


#model = FastText.load_word2vec_format(filename)
model = KeyedVectors.load_word2vec_format(filename)
              

In [4]:
model.most_similar(positive=['woman', 'king'], negative=['man'])

[('queen', 0.7515910863876343),
 ('monarch', 0.6741327047348022),
 ('princess', 0.6713887453079224),
 ('kings', 0.6698989868164062),
 ('kingdom', 0.5971318483352661),
 ('royal', 0.592106282711029),
 ('uncrowned', 0.5911506414413452),
 ('prince', 0.5909028649330139),
 ('lady', 0.5904011726379395),
 ('monarchs', 0.5884358286857605)]

# Getting the tokens 

In [6]:
words = []
for word in model.vocab:
    words.append(word)

# Printing out number of tokens available
print("Number of Tokens: {}".format(len(words)))

Number of Tokens: 999994


In [10]:
print("Dimension of a word vector: {}".format(
    len(model[words[0]])
))
## Printing out the dimension of a word vector 

Dimension of a word vector: 300


# print word in vocab

In [23]:
print("Word : {}".format(
    words[6550]
))

Word : pump


# Print out the vector of a word 

In [11]:
print("Vector components of a word: {}".format(
    model[words[0]]  
))

Vector components of a word: [ 1.0730e-01  8.9000e-03  6.0000e-04  5.5000e-03 -6.4600e-02 -6.0000e-02
  4.5000e-02 -1.3300e-02 -3.5700e-02  4.3000e-02 -3.5600e-02 -3.2000e-03
  7.3000e-03 -1.0000e-04  2.5800e-02 -1.6600e-02  7.5000e-03  6.8600e-02
  3.9200e-02  7.5300e-02  1.1500e-02 -8.7000e-03  4.2100e-02  2.6500e-02
 -6.0100e-02  2.4200e-01  1.9900e-02 -7.3900e-02 -3.1000e-03 -2.6300e-02
 -6.2000e-03  1.6800e-02 -3.5700e-02 -2.4900e-02  1.9000e-02 -1.8400e-02
 -5.3700e-02  1.4200e-01  6.0000e-02  2.2600e-02 -3.8000e-03 -6.7500e-02
 -3.6000e-03 -8.0000e-03  5.7000e-02  2.0800e-02  2.2300e-02 -2.5600e-02
 -1.5300e-02  2.2000e-03 -4.8200e-02  1.3100e-02 -6.0160e-01 -8.8000e-03
  1.0600e-02  2.2900e-02  3.3600e-02  7.1000e-03  8.8700e-02  2.3700e-02
 -2.9000e-02 -4.0500e-02 -1.2500e-02  1.4700e-02  4.7500e-02  6.4700e-02
  4.7400e-02  1.9900e-02  4.0800e-02  3.2200e-02  3.6000e-03  3.5000e-02
 -7.2300e-02 -3.0500e-02  1.8400e-02 -2.6000e-03  2.4000e-02 -1.6000e-02
 -3.0800e-02  4.3400e-

# Get word vector from dict

In [33]:
model.wv['queen']

C:\Users\DBCE\Anaconda3\envs\tf_gpu\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  """Entry point for launching an IPython kernel.


array([ 2.158e-01,  1.095e-01, -4.990e-02,  5.280e-02, -6.910e-02,
        1.357e-01, -2.257e-01, -4.010e-02, -1.270e-01,  6.280e-02,
       -3.100e-03, -2.780e-02,  9.620e-02, -5.090e-02,  1.659e-01,
       -1.456e-01,  4.300e-03, -8.580e-02,  6.750e-02, -1.441e-01,
       -1.971e-01,  2.380e-02, -1.019e-01,  2.300e-03, -1.479e-01,
       -5.790e-02, -3.480e-02,  1.964e-01,  1.310e-01,  2.600e-03,
        1.745e-01,  1.163e-01, -6.700e-03,  8.430e-02,  4.980e-02,
       -9.160e-02, -8.760e-02,  9.060e-02,  3.480e-02, -6.000e-04,
        1.479e-01, -3.700e-02, -4.900e-02, -1.296e-01,  6.300e-03,
        1.218e-01, -1.540e-02,  4.080e-02, -4.990e-02,  7.400e-03,
       -6.280e-02, -1.445e-01, -6.658e-01,  4.050e-02,  1.376e-01,
        9.190e-02,  6.400e-03,  1.542e-01,  3.450e-02, -1.420e-01,
       -6.500e-03, -3.460e-02, -1.175e-01,  1.700e-02,  9.750e-02,
        1.430e-02,  1.287e-01, -1.075e-01, -6.500e-03,  3.120e-02,
       -6.930e-02, -1.160e-01, -6.550e-02, -1.680e-02, -9.130e

# if OOV handle it  with exception and store that list
* use most_similar function to get all value above certain threashold

In [35]:
#model.wv['xrtnfsdsnknm,.46dy','NULL']
model.wv['_the']

C:\Users\DBCE\Anaconda3\envs\tf_gpu\lib\site-packages\ipykernel_launcher.py:2: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  


KeyError: "word '_the' not in vocabulary"

# Finding out similar words [default= top 10]

In [38]:
model.wv.most_similar('car')

C:\Users\DBCE\Anaconda3\envs\tf_gpu\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  """Entry point for launching an IPython kernel.


[('cars', 0.8045914769172668),
 ('automobile', 0.7667388916015625),
 ('vehicle', 0.753485918045044),
 ('Car', 0.7177952527999878),
 ('truck', 0.6989946961402893),
 ('SUV', 0.689612865447998),
 ('automobiles', 0.6783526539802551),
 ('dealership', 0.6682884097099304),
 ('garage', 0.6681075096130371),
 ('driver', 0.6541328430175781)]

In [48]:
model.wv..most_similar_cosmul('car')

C:\Users\DBCE\Anaconda3\envs\tf_gpu\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  """Entry point for launching an IPython kernel.


[('cars', 0.9022948741912842),
 ('automobile', 0.8833686113357544),
 ('vehicle', 0.8767421245574951),
 ('Car', 0.858896791934967),
 ('truck', 0.8494964838027954),
 ('SUV', 0.8448055982589722),
 ('automobiles', 0.8391755223274231),
 ('dealership', 0.8341434001922607),
 ('garage', 0.8340529799461365),
 ('driver', 0.827065646648407)]

# If word exist

In [52]:
'zootek' in model.wv.vocab

C:\Users\DBCE\Anaconda3\envs\tf_gpu\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  """Entry point for launching an IPython kernel.


False

In [53]:
# find similar to vec n chek threashold
#model.wv['zootek']

C:\Users\DBCE\Anaconda3\envs\tf_gpu\lib\site-packages\ipykernel_launcher.py:1: DeprecationWarning: Call to deprecated `wv` (Attribute will be removed in 4.0.0, use self instead).
  """Entry point for launching an IPython kernel.


KeyError: "word 'zootek' not in vocabulary"

# refer for OOV handling 2.1 Vectorising LOCAL Glove 50dims TF-iDF